internal/pkg/utils/url/url.go

```diff
@@ -23,30 +25,93 @@ func ParseQuery(query string) (m *orderedmap.OrderedMap, err error) {
 			continue
 		}
 		key, value, _ := strings.Cut(key, "=")
-		key, err1 := url.QueryUnescape(key)
-		if err1 != nil {
-			if err == nil {
-				err = err1
+		key, err = url.QueryUnescape(key)
+		if err != nil {
+			return m, err
+		}
+		value, err = url.QueryUnescape(value)
```

Comment:
I am not sure right now but this could be pricy function. It goes through the `key` or `value` and unescapes it. The cost is `n`*`n` or `n^2`. Not sure how many loops are done on `query != ""`

Reply:
the complexity is based on how many `&` characters the url have and `len(url)`. Therefore if we have 4 `&` and url has 80 characters the complexity is 4 * 2 * 80 loops. That's not trivial. I understand that it is copied just saying that `ParseQuery` is not trivial operation.

Reply:
I am maybe blind but `QueryUnescape` does loop as well. Maybe I dived too deep into the implementation just wanted to pin point that this operation is non trivial